src/order/well_founded.lean

```diff
@@ -20,6 +20,25 @@ and provide a few new definitions: `well_founded.min`, `well_founded.sup`, and `
 variables {α : Type*}
 
 namespace well_founded
+
+theorem not_gt_of_lt {r : α → α → Prop} (h : well_founded r) :
+  ∀ ⦃a b⦄, r a b → ¬ r b a
+| a := λ b hab hba, not_gt_of_lt hba hab
+using_well_founded { rel_tac := λ _ _, `[exact ⟨_, h⟩],
+                     dec_tac := tactic.assumption }
+
+protected theorem is_asymm {r : α → α → Prop} (h : well_founded r) : is_asymm α r :=
+⟨h.not_gt_of_lt⟩
+
+instance [has_well_founded α] : is_asymm α has_well_founded.r :=
+has_well_founded.wf.is_asymm
+
+protected theorem is_irrefl {r : α → α → Prop} (h : well_founded r) : is_irrefl α r :=
+@is_asymm.is_irrefl α r (h.is_asymm)
```

Comment:
Does dot notation work here at all?

Reply:
I don't believe so, since `is_asymm.is_irrefl` takes in an instance argument.